astropy/wcs/src/wcslib_wrap.c

```diff
@@ -1415,10 +1575,14 @@ PyWcsprm_p2s(
     goto exit;
   }
 
+  // Here we force a call to wcsset. Normally, WCSLIB will call wcsset automatically when
+  // calling wcsp2s, but we need to call it ourselves using PyWcsprm_cset so that we can
+  // catch cases where the units might change if e.g. they are not in SI to start with.
+  /* Force a call to wcsset here*/
+  if (self->preserve_units && PyWcsprm_cset(self, 1)) return NULL;
```

Comment:
Even for 1-line conditional, I recommend enclosing `return NULL;` in `{}`.

Reply:
Done